### PR TITLE
[choreo] add angular velocity constraint

### PR DESCRIPTION
**Pull request by bruingineer**
add angular velocity constraint to the ui
use the constraint to get a better control interval guess

part of #581

### CODE REVIEW

File: src-tauri/src/document/intervals.rs [hunk 1]
```diff
@@ -58,17 +58,28 @@ pub fn guess_control_interval_count(
...
+            let max_wheel_position_radius = config
+                .modules
+                .iter()
+                .fold(0f64, |max, &module| max.max(module.x.hypot(module.y)));
+            let max_ang_vel = max_vel / max_wheel_position_radius;
+
+            // use peak angular velocity and dtheta to calculate the time
+            // for theta to travel a cubic hermite spline profile
+            // in which the max angular velocity is not exceeded as an estimate
+            let time = (1.5 * dtheta) / max_ang_vel;
```
Comment (shueja) [inline src-tauri/src/document/intervals.rs:79]: Where does this 1.5 number come from?

**Review by shueja — CHANGES_REQUESTED** [review]
Unsure about the derivation of the 1.5 number. Otherwise lgtm

File: src-tauri/src/document/intervals.rs (hunk 1 above)
Comment (bruingineer) [reply src-tauri/src/document/intervals.rs:79]: It estimates the time needed to rotate `dtheta` without exceeding `max_ang_vel`. It comes from the cubic Hermite spline and moving around its variables.

/*
 * Proof for T = 1.5 * θ / ω:
 *
 * - The velocity function derived from the cubic Hermite spline is:
 *   v(t) = -6*θ*t^2 + 6*θ*t.
 *
 * - The peak velocity occurs at t = 0.5, where t∈[0, 1] :
 *   v(0.5) = 1.5*θ, which is the max angular velocity during the motion.
 * - To ensure this peak velocity does not exceed ω, we set:
 *   1.5 * θ = ω.
 * - The total time T needed to reach the final θ and
 *   not exceed ω is thus derived as:
 *   T = θ / (ω / 1.5) = 1.5 * θ / ω.
 *
 * This calculation ensures the peak velocity meets but does not exceed ω,
 * extending the time proportionally to meet this
 * requirement.
 */

Comment (calcmogul) [reply src-tauri/src/document/intervals.rs:79]: That proof should be included in the source code for context.

**Review by shueja — APPROVED** [review]

**Pull request merged**